streaming/java/streaming-api/src/main/java/io/ray/streaming/api/function/RichFunction.java

```diff
@@ -0,0 +1,24 @@
+package io.ray.streaming.api.function;
+
+import io.ray.streaming.api.context.RuntimeContext;
+
+/**
+ * An interface for all user-defined functions to define the life cycle methods of the
+ * functions, and access the task context where the functions get executed.
+ */
+public interface RichFunction extends Function {
+
+  /**
+   * Initialization method for user function which called before the first call to the user
+   * function.
+   * @param runtimeContext runtime context
+   */
+  void open(RuntimeContext runtimeContext);
```

Comment:
Should we add a 'init' method for initialization with a context or just merge 'init' into 'open'?

Reply:
I think `init` can be put into `open`